core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -556,6 +562,42 @@ public Object readGenericValue() throws IOException {
         return null;
     }
 
+    /**
+     * Reads a {@link NamedWriteable} from the current stream, by first reading its name and then looking for
+     * the corresponding entry in the registry by name, so that the proper object can be read and returned.
+     */
+    <C> C readNamedWriteable(String category) throws IOException {
+        String name = readString();
+        NamedWriteable<C> namedWriteable = namedWriteableRegistry.getPrototype(category, name);
+        return namedWriteable.readFrom(this);
```

Comment:
not sure I follow why that would be useful, can you explain? Isn't the default empty registry a reasonable default already? it is going to throw exception anyway if the registry is not available or if the object cannot be read. Maybe I am missing the point though :)